domain/application/service/application.go

```diff
@@ -374,12 +392,18 @@ func (s *ApplicationService) RemoveUnit(ctx context.Context, unitName string, le
 		if unitLife == life.Alive {
 			return fmt.Errorf("cannot remove unit %q: %w", unitName, applicationerrors.UnitIsAlive)
 		}
-		err = s.ensureUnitDead(ctx, unitName, leadership)
+		err = s.ensureUnitDead(ctx, unitName)
 		if err != nil {
 			return errors.Trace(err)
 		}
 		return s.st.RemoveUnitMaybeApplication(ctx, unitName)
 	})
+	if err == nil {
+		appName, _ := names.UnitApplication(unitName)
+		if err := leadershipRevoker.RevokeLeadership(appName, unitName); err != nil && !errors.Is(err, leadership.ErrClaimNotHeld) {
+			s.logger.Warningf("cannot revoke lease for dead unit %q", unitName)
```

Comment:
Should we ignore `leadership.ErrClaimNotHeld`?

Reply:
Existing logic which this has been ported from treats this as a no op which I think is fine.